src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,591 @@
+
+        # Step 6
+        # TH sends the Read Request Message to the DUT to read a global attribute from all clusters at all Endpoints
+        # AttributePath = [[Attribute = Specific Global Attribute]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(6, "Send Request Message to read one global attribute from all clusters on all endpoints")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [attribute_list])
+        returned_endpoints = read_request[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 7
+        # TH sends the Read Request Message to the DUT to read all attributes from a cluster at all Endpoints
+        # AttributePath = [[Cluster = Specific ClusterID]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(7, "Send Request Message to read all attributes from one cluster at all endpoints")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [descriptor_obj])
+        returned_endpoints = read_request[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 8
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters at one Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(8, "Send Request Message to read all attributes from all clusters at one endpoint")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [0])
+        returned_endpoints = read_request[0].keys()
+
+        # Check if chip.clusters.Objects.Descriptor is in output
+        asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+        # Check if ServerList is in nested output
+        asserts.assert_in(server_list_attr, read_request[0][descriptor_obj], "ServerList not in output")
```

Comment:
Can you add a check that the returned clusters match the server list?

Reply:
Added and will show up in next commit